test_cb/test_cb_regression_verilog_sim.py

```diff
@@ -1,45 +1,48 @@
 import magma as m
 from cb.cb_magma import define_cb
 from common.genesis_wrapper import run_genesis
-from common.util import skip_unless_irun_available
+from common.util import irun_available, iverilog_available
 from common.irun import irun
+from common.iverilog import iverilog
+import pytest
+import os
 
 
-def run_ncsim_regression(params):
+def run_verilog_regression(params):
     # Magma version.
     magma_cb = define_cb(**params)
     m.compile(f"./{magma_cb.name}", magma_cb, output="coreir-verilog")
 
     # Genesis version.
     genesis_outfile = run_genesis("cb", "cb/genesis/cb.vp", params)
 
-    # Run ncsim.
     files = [f"test_cb/{magma_cb.name}_tb.v",  # test bench file
              genesis_outfile,
              f"{magma_cb.name}.v"]
-    res = irun(files)
-    assert res
-
-
-@skip_unless_irun_available
-def test_16_10_111110111_1_7():
-    params = {
-        "width": 16,
-        "num_tracks": 10,
-        "feedthrough_outputs": "1111101111",
-        "has_constant": 1,
-        "default_value": 7,
-    }
-    run_ncsim_regression(params)
-
-
-@skip_unless_irun_available
-def test_7_8_11111111_0_0():
-    params = {
+    if irun_available:
+        # Run ncsim.
+        assert irun(files)
+    else:
+        assert iverilog_available, "Should be true based on pytest skip marker"
+        assert iverilog(files)
+
+
+@pytest.mark.skipif(not irun_available and not iverilog_available,
```

Comment:
If we add a `verilog_compiler_available` function, we can just call that here, rather than taking the or of these conditions.